### PR TITLE
Test PodDisruptionBudgets

### PR DESCRIPTION
**Pull request by ramperher**
build-depends: https://github.com/dci-labs/example-cnf-config/pull/96

- This is to fix pod-disruption-budget tnf test
- Config provided handles the case where we only have one replica deployed, where we don't want to have a node that is impossible to drain

### CODE REVIEW

**Comment by ramperher** [comment]
pod-disruption-budget test is now passing: https://www.distributed-ci.io/jobs/41f63c56-44e3-43c8-85c0-3535751e02f6/tests/5eb2475b-4b87-4a35-9fae-a71026c9cd84 , this is ready for review

File: testpmd-operator/roles/testpmd/tasks/main.yml [hunk 1]
```diff
@@ -80,3 +80,12 @@
   kubernetes.core.k8s:
     state: present
     definition: "{{ lookup('template', 'deployment.yml') }}"
+
+# if replica = 1, recommended not to set maxUnavailable to 0 or 0% and minAvailable to 100% for upgrades
+# https://github.com/operator-framework/operator-lifecycle-manager/blob/master/doc/design/adding-pod-disruption-budgets.md#limitations-on-pod-disruption-budgets
+# TODO (in a new patch): update PDB during upgrades to avoid issues during node draining, something like:
+# https://medium.com/@tamber/solution-avoid-kubernetes-openshift-node-drain-failure-due-to-active-poddisruptionbudget-df68efed2c4f
+- name: Create PodDisruptionBudget for LoadBalancer pod
```
Comment (manurodriguez) [inline testpmd-operator/roles/testpmd/tasks/main.yml:88]: Is this for the LoadBalancer pod or should we change the task name?

File: trex-operator/roles/server/tasks/main.yml [hunk 2]
```diff
@@ -34,6 +34,15 @@
     state: present
     definition: "{{ lookup('template', 'deployment.yml') }}"
 
+# if replica = 1, recommended not to set maxUnavailable to 0 or 0% and minAvailable to 100% for upgrades
+# https://github.com/operator-framework/operator-lifecycle-manager/blob/master/doc/design/adding-pod-disruption-budgets.md#limitations-on-pod-disruption-budgets
+# TODO (in a new patch): update PDB during upgrades to avoid issues during node draining, something like:
+# https://medium.com/@tamber/solution-avoid-kubernetes-openshift-node-drain-failure-due-to-active-poddisruptionbudget-df68efed2c4f
+- name: Create PodDisruptionBudget for LoadBalancer pod
```
Comment (manurodriguez) [inline trex-operator/roles/server/tasks/main.yml:41]: Same here about the LoadBalancer pod, maybe we should say that is for the trex pod?

**Comment by manurodriguez** [comment]
> pod-disruption-budget test is now passing: https://www.distributed-ci.io/jobs/41f63c56-44e3-43c8-85c0-3535751e02f6/tests/5eb2475b-4b87-4a35-9fae-a71026c9cd84 , this is ready for review

Just one question: I checked the job log and I see the task [Create PodDisruptionBudget for controller-manager pods](https://www.distributed-ci.io/jobs/41f63c56-44e3-43c8-85c0-3535751e02f6/jobStates?sort=date&task=0b8d4a85-951a-46ac-85b6-7935f2db0ee9) that corresponds to change in https://github.com/dci-labs/example-cnf-config/pull/96. But I do not see where the `Create PodDisruptionBudget for *` tasks of this PR#61 are executed. Are those not tracked during the job execution? Maybe it's in another place :thinking:

File: testpmd-operator/roles/testpmd/tasks/main.yml (hunk 1 above)
Comment (ramperher) [reply testpmd-operator/roles/testpmd/tasks/main.yml]: that's true, fixed

File: trex-operator/roles/server/tasks/main.yml (hunk 2 above)
Comment (ramperher) [reply trex-operator/roles/server/tasks/main.yml]: done

**Review by ramperher — COMMENTED** [review]
Thanks @manurodriguez for catching these typos! Now they're fixed

**Comment by ramperher** [comment]
> > pod-disruption-budget test is now passing: https://www.distributed-ci.io/jobs/41f63c56-44e3-43c8-85c0-3535751e02f6/tests/5eb2475b-4b87-4a35-9fae-a71026c9cd84 , this is ready for review
> 
> Just one question: I checked the job log and I see the task [Create PodDisruptionBudget for controller-manager pods](https://www.distributed-ci.io/jobs/41f63c56-44e3-43c8-85c0-3535751e02f6/jobStates?sort=date&task=0b8d4a85-951a-46ac-85b6-7935f2db0ee9) that corresponds to change in [dci-labs/example-cnf-config#96](https://github.com/dci-labs/example-cnf-config/pull/96). But I do not see where the `Create PodDisruptionBudget for *` tasks of this PR#61 are executed. Are those not tracked during the job execution? Maybe it's in another place 🤔

You're right, in the job, we can only see the task from example-cnf-config, which is directly included in the hooks because, for the controller-manager pods, we need to do it from an external point of view.

For the other pods (LoadBalancer, TestPMD and TRex), you can see how they're created in the controller-manager logs, e.g. testpmd-lb-operator-controller-manager-5578fb6c98-s2fkw.log
:

```
...
--------------------------- Ansible Task StdOut -------------------------------

TASK [loadbalancer : Create PodDisruptionBudget for LoadBalancer pod] **********
[1;30mtask path: /opt/ansible/roles/loadbalancer/tasks/main.yml:67[0m

-------------------------------------------------------------------------------
{"level":"info","ts":"2024-03-04T18:55:59Z","logger":"proxy","msg":"Cache miss: policy/v1, Kind=PodDisruptionBudget, example-cnf/lb-app-pdb-min"}
{"level":"info","ts":"2024-03-04T18:55:59Z","logger":"proxy","msg":"Injecting owner reference"}
{"level":"info","ts":"2024-03-04T18:55:59Z","logger":"proxy","msg":"Watching child resource","kind":"policy/v1, Kind=PodDisruptionBudget","enqueue_kind":"examplecnf.openshift.io/v1, Kind=LoadBalancer"}
{"level":"info","ts":"2024-03-04T18:55:59Z","msg":"Starting EventSource","controller":"loadbalancer-controller","source":"kind source: *unstructured.Unstructured"}
{"level":"info","ts":"2024-03-04T18:55:59Z","logger":"proxy","msg":"Cache miss: policy/v1, Kind=PodDisruptionBudget, example-cnf/lb-app-pdb-max"}
{"level":"info","ts":"2024-03-04T18:55:59Z","logger":"proxy","msg":"Injecting owner reference"}
{"level":"info","ts":"2024-03-04T18:55:59Z","logger":"runner","msg":"Ansible-runner exited successfully","job":"2591084995821321510","name":"lb","namespace":"example-cnf"}

----- Ansible Task Status Event StdOut (examplecnf.openshift.io/v1, Kind=LoadBalancer, lb/example-cnf) -----

...
```

**Review by manurodriguez — APPROVED** [review]
Thanks for the clarification, LGTM

**Pull request merged**